### PR TITLE
SMF sends PDU Session Establishment Reject to UE when it fails to release the old session with duplicated PSI

### PR DESCRIPTION
**Pull request by mariakagi**

### CODE REVIEW

File: internal/gmm/handler.go [hunk 1]
```diff
@@ -125,19 +125,16 @@ func transport5GSMMessage(ue *context.AmfUe, anType models.AccessType,
 				}
 				ue.GmmLog.Warningf("Duplicated PDU session ID[%d]", pduSessionID)
 				smContext.SetDuplicatedPduSessionID(true)
-				response, errResponse, _, err := consumer.SendUpdateSmContextRequest(smContext, updateData, nil, nil)
+				response, _, _, err := consumer.SendUpdateSmContextRequest(smContext, updateData, nil, nil)
 				if err != nil {
+					smContext.SetDuplicatedPduSessionID(false)
```
Comment (ianchen0119) [inline internal/gmm/handler.go:130]: I think the duplicated attribute should be `true` if `SendUpdateSmContextRequest()` gets error.

Comment (mariakagi) [reply internal/gmm/handler.go:130]: I think it is better to unset the duplicated attribute when `SendUpdateSmContextRequest()` gets error for the following reasons: 
* When the network initiates PDU session release (e.g. 4.3.4.2 1e in TS 23.502) for some reason after `SendUpdateSmContextRequest()` fails, if the attribute is true, AMF tries to create the new SMContext by invocation of Nsmf_PDUSession_SMContextStatusNotify. If the attribute is false, PDU session release completes and no new session is established.
* If `SendUpdateSmContextRequest()` gets error, AMF doesn't send response to UE. On the expiry of the timer T3580 UE retransmits the PDU Session Establishment Request, the attribute can be set again, so there is no problem.

Comment (ianchen0119) [reply internal/gmm/handler.go:130]: @mariakagi 
I got it, thanks for your explanation!

Comment (tim-ywliu) [reply internal/gmm/handler.go:130]: @mariakagi 

It's better to do local release smContext when error or error response happened:
```
				if err != nil {
					ue.GmmLog.Errorf("SendUpdateSmContextRequest err: %v, local release SmContext[%d]",
						err, pduSessionID)
					ue.DeleteSmContext(pduSessionID)
					return err
				} else if response == nil {
					ue.GmmLog.Errorf("SendUpdateSmContextRequest rsp is nil, local release SmContext[%d]",
						pduSessionID)
					ue.DeleteSmContext(pduSessionID)
				} else if response != nil {

```

Comment (mariakagi) [reply internal/gmm/handler.go:130]: @tim-ywliu 
Does `ue.DeleteSmContext(pduSessionID)` mean `ue.SmContextList.Delete(pduSessionID)` ?

Comment (tim-ywliu) [reply internal/gmm/handler.go:130]: @mariakagi 

Yes.

Comment (mariakagi) [reply internal/gmm/handler.go:130]: Your suggestion looks good.
But when `if err != nil`, `"SendUpdateSmContextRequest err: ..."` is output twice.

Comment (tim-ywliu) [reply internal/gmm/handler.go:130]: @mariakagi 

Ya, the error log here can be removed.

Comment (mariakagi) [reply internal/gmm/handler.go]: I made a modification to delete smContext and revise the text output to log.

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**